RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -177,48 +283,151 @@ StatusCode AugmentClustersFCCee::execute([[maybe_unused]] const EventContext &ev
+        double weightLog = std::max(0., m_thetaRecalcLayerWeights[k][layer] + log(eCell / sumEnLayer[layer]));  // TODO layer+startPositionToFill ?
         TVector3 v = TVector3(cell->getPosition().x, cell->getPosition().y, cell->getPosition().z);
         double theta = v.Theta();
         double phi = v.Phi();
+
         // for clusters that are around the -pi<->pi transition, we want to avoid averaging
         // over phi values that might differ by 2pi. in that case, for cells with negative
         // phi we add two pi, so that we average phi values all close to pi
-        if (isClusterPhiNearPi && phi < 0.)
+        if (isClusterPhiNearPi && phi < 0.) {
           phi += TMath::TwoPi();
+        }
+        if (systemID == 4 && isResetModuleID && module_id > nModules_EMB/2) {
+          module_id -= nModules_EMB;  // transition of module ID
+        }
+
         if (m_thetaRecalcLayerWeights[k][layer]<0)
           sumThetaLayer[layer+startPositionToFill] += (eCell * theta);
         else
           sumThetaLayer[layer+startPositionToFill] += (weightLog * theta);
         sumWeightLayer[layer+startPositionToFill] += weightLog;
         sumPhiLayer[layer+startPositionToFill] += (eCell * phi);
-      }
+
+        // do pi0/photon shape var only for EMB
+        if (m_do_pi0_photon_shapeVar && systemID == 4) {
+          // E, theta_id, and module_id of cells in layer
+          vec_E_cell_layer[layer+startPositionToFill].push_back(eCell);
+          vec_theta_cell_layer[layer+startPositionToFill].push_back(theta_id);
+          vec_module_cell_layer[layer+startPositionToFill].push_back(module_id);
+          // sum them for width calculation
+          theta2_E_layer[layer+startPositionToFill] += theta_id * theta_id * eCell;
+          theta_E_layer[layer+startPositionToFill] += theta_id * eCell;
+          module2_E_layer[layer+startPositionToFill] += module_id * module_id * eCell;
+          module_E_layer[layer+startPositionToFill] += module_id * eCell;
+        }
+      }  // end of loop over cells
+    }  // end of loop over each system / readout
+
+    std::vector<std::pair<std::vector<int>, std::vector<double>>> theta_E_pair;
+    // local maxima (could be more than one) and the corresponding theta
+    std::vector<std::vector<double>> local_E_Max(numLayersTotal, std::vector<double>());
+    std::vector<std::vector<int>> local_E_Max_theta(numLayersTotal, std::vector<int>());
+
+    std::vector<double> E_cell_Max(numLayersTotal, 0.);
+    std::vector<double> E_cell_secMax(numLayersTotal, 0.);
+    std::vector<double> E_cell_Max_theta(numLayersTotal, 0.);
+    std::vector<double> E_cell_secMax_theta(numLayersTotal, 0.);
+    std::vector<double> E_cell_Min(numLayersTotal, std::numeric_limits<double>::max());
+
+    startPositionToFill = 0;
+    for (size_t k = 0; k < m_readoutNames.size(); k++) {
+      if (!m_do_pi0_photon_shapeVar)    break;
+      if (m_systemIDs[k] != 4)    continue;    // do pi0/photon shape var only for EMB
```

Comment:
`mydetector.constantAsDouble("DetID_ECAL_Barrel");`  Actually since it appears many times, you could just define a `systemID_EMB = mydetector.constantAsDouble("DetID_ECAL_Barrel");` once.